<!-- This comment will put IE 6, 7 and 8 in quirks mode -->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>SILVERCODERS DocToText: doctotext::Exception Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javaScript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body onload='searchBox.OnSelectItem(0);'>
<!-- Generated by Doxygen 1.6.1 -->
<script type="text/javascript"><!--
var searchBox = new SearchBox("searchBox", "search",false,'Search');
--></script>
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <img id="MSearchSelect" src="search/search.png"
             onmouseover="return searchBox.OnSearchSelectShow()"
             onmouseout="return searchBox.OnSearchSelectHide()"
             alt=""/>
        <input type="text" id="MSearchField" value="Search" accesskey="S"
             onfocus="searchBox.OnSearchFieldFocus(true)" 
             onblur="searchBox.OnSearchFieldFocus(false)" 
             onkeyup="searchBox.OnSearchFieldChange(event)"/>
        <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
        </div>
      </li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath"><b>doctotext</b>::<a class="el" href="classdoctotext_1_1Exception.html">Exception</a>
  </div>
</div>
<div class="contents">
<h1>doctotext::Exception Class Reference</h1><!-- doxytag: class="doctotext::Exception" -->
<p><code>#include &lt;<a class="el" href="exception_8h_source.html">exception.h</a>&gt;</code></p>

<p><a href="classdoctotext_1_1Exception-members.html">List of all members.</a></p>
<table border="0" cellpadding="0" cellspacing="0">
<tr><td colspan="2"><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classdoctotext_1_1Exception.html#a5db39ff707c5408549a705136c760a4b">Exception</a> (const std::string &amp;first_error_message)  throw ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="ae10dd7e1e1567a4b3fe2a0e105dd8c5b"></a><!-- doxytag: member="doctotext::Exception::Exception" ref="ae10dd7e1e1567a4b3fe2a0e105dd8c5b" args="(const Exception &amp;ex)" -->
&nbsp;</td><td class="memItemRight" valign="bottom"><b>Exception</b> (const <a class="el" href="classdoctotext_1_1Exception.html">Exception</a> &amp;ex)  throw ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a8409e8467bfa6309dc6e6f2dc3219e12"></a><!-- doxytag: member="doctotext::Exception::operator=" ref="a8409e8467bfa6309dc6e6f2dc3219e12" args="(const Exception &amp;ex)" -->
<a class="el" href="classdoctotext_1_1Exception.html">Exception</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><b>operator=</b> (const <a class="el" href="classdoctotext_1_1Exception.html">Exception</a> &amp;ex)  throw ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a32cb3156f2adfd140c0bb5b6cd7ad78f"></a><!-- doxytag: member="doctotext::Exception::what" ref="a32cb3156f2adfd140c0bb5b6cd7ad78f" args="()" -->
const char *&nbsp;</td><td class="memItemRight" valign="bottom"><b>what</b> ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">std::string&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classdoctotext_1_1Exception.html#ab88499ca7184b914041eed0a38c82b9f">getBacktrace</a> ()</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classdoctotext_1_1Exception.html#ac612f8be8b7c1d0f86f9bc33548f37c2">appendError</a> (const std::string &amp;error_message)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">std::list&lt; std::string &gt;::iterator&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classdoctotext_1_1Exception.html#a48a13fbb9923cfa41dbde8a5b32a6b46">getErrorIterator</a> () const </td></tr>
<tr><td class="memItemLeft" align="right" valign="top">size_t&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classdoctotext_1_1Exception.html#a22f660e0d2a40bba5721d18448c63652">getErrorCount</a> () const </td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>This class is implementation of std::exception, which is used by DocToText. In this implementation, errors can be formed in "stack".</p>
<p>In order to create exception just call: </p>
<div class="fragment"><pre class="fragment">                <span class="keywordflow">throw</span> Exception(<span class="stringliteral">&quot;First error&quot;</span>);
</pre></div><p>You can catch exception and add one more error: </p>
<div class="fragment"><pre class="fragment">                <span class="keywordflow">catch</span> (Exception&amp; ex)
                {
                        ex.appendError(<span class="stringliteral">&quot;Next error message&quot;</span>);
                        <span class="keywordflow">throw</span>;
                }
</pre></div><p>or you can catch exception and get "backtrace": </p>
<div class="fragment"><pre class="fragment">                <span class="keywordflow">catch</span> (Exception&amp; ex)
                {
                        std::cerr &lt;&lt; ex.getBacktrace();
                }
</pre></div> <hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a5db39ff707c5408549a705136c760a4b"></a><!-- doxytag: member="doctotext::Exception::Exception" ref="a5db39ff707c5408549a705136c760a4b" args="(const std::string &amp;first_error_message)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">doctotext::Exception::Exception </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>first_error_message</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td>  throw ()</td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>The constructor. </p>
<dl><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>first_error_message</em>&nbsp;</td><td>first error message (gives information about cause of an error). </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="ac612f8be8b7c1d0f86f9bc33548f37c2"></a><!-- doxytag: member="doctotext::Exception::appendError" ref="ac612f8be8b7c1d0f86f9bc33548f37c2" args="(const std::string &amp;error_message)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void doctotext::Exception::appendError </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>error_message</em></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Adds one more error message. </p>

</div>
</div>
<a class="anchor" id="ab88499ca7184b914041eed0a38c82b9f"></a><!-- doxytag: member="doctotext::Exception::getBacktrace" ref="ab88499ca7184b914041eed0a38c82b9f" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string doctotext::Exception::getBacktrace </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>returns a string with all error messages. Each error message is separated by "\n". Suppose we have thrown an exception: </p>
<div class="fragment"><pre class="fragment">                                <span class="keywordflow">throw</span> Exception(<span class="stringliteral">&quot;First error message&quot;</span>);
</pre></div><p> Next, we have added one more error: </p>
<div class="fragment"><pre class="fragment">                                ex.appendError(<span class="stringliteral">&quot;Second error message&quot;</span>);
</pre></div><p> In the result getBacktrace will return a string: "First error message\nSecond error message\n" </p>

</div>
</div>
<a class="anchor" id="a22f660e0d2a40bba5721d18448c63652"></a><!-- doxytag: member="doctotext::Exception::getErrorCount" ref="a22f660e0d2a40bba5721d18448c63652" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">size_t doctotext::Exception::getErrorCount </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const</td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Returns a number of error messages </p>

</div>
</div>
<a class="anchor" id="a48a13fbb9923cfa41dbde8a5b32a6b46"></a><!-- doxytag: member="doctotext::Exception::getErrorIterator" ref="a48a13fbb9923cfa41dbde8a5b32a6b46" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::list&lt;std::string&gt;::iterator doctotext::Exception::getErrorIterator </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const</td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>returns a iterator to the first error message </p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="exception_8h_source.html">exception.h</a></li>
</ul>
</div>
<!--- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&nbsp;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&nbsp;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&nbsp;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&nbsp;</span>Enumerations</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<hr size="1"/><address style="text-align: right;"><small>Generated on 9 Feb 2017 for SILVERCODERS DocToText by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.6.1 </small></address>
</body>
</html>
